docs/DeveloperGuide.md

```diff
@@ -4,10 +4,137 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architecture](images\Architecture.png)
```

Comment:
An excellent high-level class diagram! Simple, clear and straight to the point. Good for understanding.